### ISSUE TITLE
In-Place Activated BatchNorm for Memory-Optimized Training of DNNs

### ISSUE THREAD

**Issue by lisosia** [lisosia/cv_knowledge#4]
## 一言でいうと
activationが逆変換可能(~= 一様増加, LeakeyReluなど）の場合、BatchNormを少し修正すれば、batchnorm+reslu後のxの保存を省略でき、最大50%メモリ省略できる。

### 論文リンク
https://arxiv.org/pdf/1712.02616.pdf

### 著者/所属機関
Mapillary Research

### 投稿日付(yyyy/MM/dd)
2018/10/26

## 結果
![スクリーンショット 2020-07-08 08-34-31](https://user-images.githubusercontent.com/8558911/86855705-dd449000-c0f5-11ea-9f34-4f2778b62d80.png)

## コメント
batchsize同条件ならbatchnormより劣化する？  
また、batchsize同条件の精度比較や、メモリ結局いくら使うかの表グラフがないのが解せない。

実装：
https://github.com/mapillary/inplace_abn
